public/app/features/annotations/standardAnnotationSupport.ts

```diff
@@ -0,0 +1,188 @@
+import {
+  DataFrame,
+  AnnotationQuery,
+  AnnotationSupport,
+  PanelData,
+  transformDataFrame,
+  FieldType,
+  Field,
+  KeyValue,
+  AnnotationEvent,
+  AnnotationEventMappings,
+  getFieldDisplayName,
+  AnnotationEventFieldSource,
+} from '@grafana/data';
+
+import isString from 'lodash/isString';
+
+export const standardAnnotationSupport: AnnotationSupport = {
+  /**
+   * Assume the stored value is standard model.
+   */
+  prepareAnnotation: (json: any) => {
+    if (isString(json?.query)) {
+      const { query, ...rest } = json;
+      return {
+        ...rest,
+        target: {
+          query,
+        },
+        mappings: {},
+      };
+    }
+    return json as AnnotationQuery;
+  },
+
+  /**
+   * Convert the stored JSON model and environment to a standard datasource query object.
+   * This query will be executed in the datasource and the results converted into events.
+   * Returning an undefined result will quietly skip query execution
+   */
+  prepareQuery: (anno: AnnotationQuery) => anno.target,
+
+  /**
+   * When the standard frame > event processing is insufficient, this allows explicit control of the mappings
+   */
+  processEvents: (anno: AnnotationQuery, data: DataFrame) => {
+    return getAnnotationsFromFrame(data, anno.mappings);
+  },
+};
+
+/**
+ * Flatten all panel data into a single frame
+ */
+export function singleFrameFromPanelData(rsp: PanelData): DataFrame | undefined {
+  if (!rsp?.series?.length) {
+    return undefined;
+  }
+  if (rsp.series.length === 1) {
+    return rsp.series[0];
+  }
+
+  return transformDataFrame(
+    [
+      {
+        id: 'seriesToColumns',
+        options: { byField: 'Time' },
+      },
+    ],
+    rsp.series
+  )[0];
+}
+
+interface AnnotationEventFieldSetter {
+  key: keyof AnnotationEvent;
+  field?: Field;
+  text?: string;
+  regex?: RegExp;
+  split?: string; // for tags
+}
+
+export interface AnnotationFieldInfo {
+  key: keyof AnnotationEvent;
+
+  split?: string;
+  field?: (frame: DataFrame) => Field | undefined;
+  placeholder?: string;
+  help?: string;
+}
+
+export const annotationEventNames: AnnotationFieldInfo[] = [
+  {
+    key: 'time',
+    field: (frame: DataFrame) => frame.fields.find(f => f.type === FieldType.time),
+    placeholder: 'time, or the first time field',
+  },
+  { key: 'timeEnd', help: 'When this field is defined, the annotation will be treated as a range' },
+  {
+    key: 'title',
+  },
+  {
+    key: 'text',
+    field: (frame: DataFrame) => frame.fields.find(f => f.type === FieldType.string),
+    placeholder: 'text, or the first text field',
```

Comment:
currently, the preferred annotation event structure only has a text & tags field, the title field was deprecated many years ago (but I see that it still there in some annotation editors), Thought we deprecated it and only showed it if was configured but seems like that is not the case. Think Elasticsearch & InfluxDB only show the title field if it's already defined in the annotation (for backward compatibility).

Reply:
As for rendering text & title just get's appended in the annotation tooltip body

Reply:
So what should we expose?  Just text?


FYI -- plugins are currently blindly filling in as many of the fields as possible.  There is no indication anything is deprecated

Reply:
yea, I know. This was removed a long time ago before we had types and deprecation warnings. 

I think we should only expose text for now as both title and text are just concatenated anyway when we render annotations.